crypto/crypto/src/noise.rs

```diff
@@ -645,8 +645,8 @@ impl NoiseSession {
             .encrypt_in_place_detached(nonce, b"", message)
             .map_err(|_| NoiseError::Encrypt)?;
 
-        // increment nonce
-        self.write_nonce += 1;
+        // increment nonce, which won't exceed the maximum u64 value in practice
```

Comment:
This is a claim without evidence. Why?

Reply:
While this is true (because the nonce is initialized to 0 and never reaches `u64::max`), if it ever becomes false then this change will introduce a bug. I would propose another change: since this function returns a `Result`, let's use a `checked_add` and return an error if the addition wraps around.

Reply:
As suggested by @huitseeker, in order to keep this PR tag-specific, I've separated new bound checks from this PR to another PR #5540.